tests/smoke/smoke_test.go

```diff
@@ -1,13 +1,275 @@
+					}
+					port = i.Ports[0].Port.IntValue()
+					if p == "TCP" && port == 3306 {
+						flag_i = 1
+					}
+				}
+			}
+		}
+		if flag == 1 && flag_i == 1 {
+			return flag, flag_i
+		}
+	}
+	return flag, flag_i
+}
+
+// WordpressPortForward enable port forwarding for wordpress
+func WordpressPortForward() error {
+	if stopChan != nil {
+		log.Error().Msgf("wordpress port forward is already in progress")
+		return errors.New("wordpress port forward is already in progress")
+	}
+	ns := "wordpress-mysql"
+	pods, err := util.K8sGetPods("^wordpress-..........-.....$", ns, nil, 0)
+	if err != nil {
+		log.Printf("could not get wordpress pods assuming process mode")
+		return nil
+	}
+	if len(pods) != 1 {
+		log.Error().Msgf("len(pods)=%d", len(pods))
+		return errors.New("expecting one wordpress pod only")
+	}
+	log.Printf("found wordpress pod:[%s]", pods[0])
+	c, err := util.K8sPortForward(util.PortForwardOpt{
+		LocalPort:   8000,
+		RemotePort:  80,
+		ServiceName: pods[0],
+		Namespace:   ns})
+	if err != nil {
+		log.Error().Msgf("could not do wordpress portforward Error=%s", err.Error())
+		return err
+	}
+	stopChan = c
+	return nil
+}
+
+// WordpressPortForwardStop stop wordpress port forwarding
+func WordpressPortForwardStop() {
+	if stopChan == nil {
+		return
+	}
+	close(stopChan)
+	stopChan = nil
+}
+
+var _ = BeforeSuite(func() {
+	// install discovery-engine
+	_, err := util.Kubectl(fmt.Sprintf("apply -f https://raw.githubusercontent.com/kubearmor/discovery-engine/dev/deployments/k8s/deployment.yaml"))
+	Expect(err).To(BeNil())
+	// check discovery-engine pod status
+	checkPod("discovery-engine-",
+		"container.apparmor.security.beta.kubernetes.io/discovery-engine: localhost/kubearmor-accuknox-agents-discovery-engine-discovery-engine", "accuknox-agents")
+
+	//install wordpress-mysql app
+	err = util.K8sApply([]string{"res/wordpress-mysql-deployment.yaml"})
+	Expect(err).To(BeNil())
+	//check wordpress pod status
+	checkPod("wordpress-",
+		"container.apparmor.security.beta.kubernetes.io/wordpress: localhost/kubearmor-wordpress-mysql-wordpress-wordpress", "wordpress-mysql")
+	//check mysql pod status
+	checkPod("mysql-",
+		"container.apparmor.security.beta.kubernetes.io/mysql: localhost/kubearmor-wordpress-mysql-mysql-mysql", "wordpress-mysql")
+
+	// delete all KSPs
+	err = util.DeleteAllKsp()
+	Expect(err).To(BeNil())
+
+	// enable kubearmor port forwarding
+	err = util.KubearmorPortForward()
+	Expect(err).To(BeNil())
+
+	// enable wordpress port forwarding
+	err = WordpressPortForward()
+	Expect(err).To(BeNil())
+
+})
+
+var _ = AfterSuite(func() {
+	util.KubearmorPortForwardStop()
+	WordpressPortForwardStop()
+})
+
+func discoversyspolicy(ns string, l string, rules []string, maxcnt int) (types.KubeArmorPolicy, error) {
+	policy := types.KubeArmorPolicy{}
+	var err error
+	for cnt := 0; cnt < maxcnt; cnt++ {
+		cmd, err := exec.Command("karmor", "discover", "-n", ns, "-l", l, "-f", "json").Output()
```

Comment:
@Prateeknandle I think we should use api calls from karmor instead of using exec. Ref: https://github.com/kubearmor/kubearmor-client/issues/241#issue-1531696567

Do you think it's doable to refactor discover package in karmor client?

Reply:
> Do you think it's doable to refactor discover package in karmor client?

yes, so basically we need to return the discovered policy output also with the print statement which already exists?

Reply:
We can convert the discover function to return the policies, and then make a new function in the discover.go code where it calls the discover function to print to Stdout. This way the original karmor summary command is not broken, as well as we can call the discover function here.

Reply:
you mean convert [this](https://github.com/kubearmor/kubearmor-client/blob/main/discover/discover.go#L43) func to return the policies?

Reply:
yeah the ConvertPolicy function should return. And then all the marshalling and output can be done in the Policy() function.

Reply:
@PrimalPimmy this can't be done right now, there are some blockers to implement this, @rksharma95 and I discussed to implement it later